src/features/surveys/utils/prepareSurveyApiSubmission.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  ZetkinSurveyApiSubmission,
+  ZetkinSurveyQuestionResponse,
+  ZetkinSurveySignaturePayload,
+} from 'utils/types/zetkin';
+
+export default function prepareSurveyApiSubmission(
+  formData: NodeJS.Dict<string | string[]>,
+  isLoggedIn?: boolean
+): ZetkinSurveyApiSubmission {
+  const responses: ZetkinSurveyQuestionResponse[] = [];
+  const responseEntries = Object.fromEntries(
+    Object.entries(formData).filter(([name]) =>
+      name.match(/^[0-9]+\.(options|text)+$/)
```

Comment:
I love regexes (in a nerdy, not necessarily healthy way) but I'm also glad we have tests to confirm that this works. 😊 

However, as I was looking at this regex more closely while writing this shoutout, I realized there is actually a "bug" here. Your regex will match `123.texttext`, `123.optionsoptions`, `123.optionstextoptionstexttext` etc, because of that final `+`. I believe you don't need that, right?

I guess it doesn't really matter in terms of getting this code to work (because the form would never contain fields like that) but I think it's best that the regex reflects what we actually want to do, so as to not confuse anyone reading it.

Reply:
Fixed! Nice spot. Bit of a silly use of regex really.